Source/WebCore/css/CSSGradientValue.cpp

```diff
@@ -36,20 +36,18 @@
 
 namespace WebCore {
 
-Vector<StyleGradientImageStop> CSSGradientValue::computeStops(Style::BuilderState& state) const
+static inline std::optional<StyleColor> computeStyleGradientImageStopColor(const RefPtr<CSSPrimitiveValue>& color, Style::BuilderState& state)
 {
-    Vector<StyleGradientImageStop> stops;
-    stops.reserveInitialCapacity(m_stops.size());
-
-    for (auto& stop : m_stops) {
-        stops.uncheckedAppend({
-            stop.color,
-            stop.position,
-            stop.color ? state.colorFromPrimitiveValueWithResolvedCurrentColor(*stop.color) : Color()
-        });
-    }
+    if (!color)
+        return std::nullopt;
+    return state.colorFromPrimitiveValue(*color);
+}
 
-    return stops;
+Vector<StyleGradientImageStop> CSSGradientValue::computeStops(Style::BuilderState& state) const
```

Comment:
So sad to have to utter `StyleGradientImageStop` three times here. Glad we don’t have to do it four times, though!

Reply:
We don't have to if we don't want to!

```c++
static inline std::optional<StyleColor> computeStyleColor(const RefPtr<CSSPrimitiveValue>& color, Style::BuilderState& state)
{
    if (!color)
        return std::nullopt;
    return state.colorFromPrimitiveValue(*color);
}

decltype(auto) CSSGradientValue::computeStops(Style::BuilderState& state) const
{
    return m_stops.map([&] (auto& stop) -> StyleGradientImageStop {
        return { computeStyleColor(stop.color, state), stop.position };
    });
}
```

This works just as well :).